### DISCUSSION TITLE
How to erase the pushed wrong commit entirely?

### DISCUSSION THREAD

**Discussion by MemphisMeng** [community/community#35474] [Code Security]
I pushed a commit to a branch with my password which was potentially threatening my organization. My solution is
```
git reset --hard HEAD~1
```
+
```
git push --force origin
```
It worked only a half because on the one hand, it did remove the commit record my log tree since I got `fatal: bad object xxxxxxxx` when doing `git log xxxxxxxx` ("xxxxxxxxx" is the commit's full ID); on the other, https://github.com/my-org/my-repo/tree/xxxxxxxx can still be opened which means my password is still visible online.

So does GitHub truly do so on purpose and we users have no way to get 100% erased?

**Comment by airtower-luna** [comment]
In short: GitHub doesn't actively delete unreferenced commits, so it may stick around. You can [contact GitHub support](https://support.github.com/request) and ask them to delete the sensitive commit from the cache, but **you should consider the password compromised** and change it anyway. :warning: 

See the full documentation, especially the warning box near the top: https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/removing-sensitive-data-from-a-repository
